File: task2/.py
def normalize_email_py(email):
    parts = email.split('@')
    if len(parts) != 2:
        return ""  # Invalid email format
    username = parts[0]
    domain = parts[1]

    normalized_username = ""
    for char in username:
        if char == '.':
            continue
        elif char == '+' or char == '*':
            break
        else:
            normalized_username += char
    return normalized_username + "@" + domain

def count_unique_emails_py(emails):
    unique_emails = set()
    for email in emails:
        unique_emails.add(normalize_email_py(email))
    return len(unique_emails)

emails1 = ["mar.pha+science@corp.nstu.ru", "marpha+scie.nce@corp.nstu.ru", "marph.a+s.c.i.e.n.c.e+@corp.nstu.ru"]
print("Example 1 Output:", count_unique_emails_py(emails1))

emails2 = ["mar.pha+science@co.rp.nstu.ru", "marpha+scie.nce@corp.nstu.ru", "marph.a+s.c.i.e.n.c.e+@corp.nstu.ru"]
print("Example 2 Output:", count_unique_emails_py(emails2))

File: task2/.java
import java.util.HashSet;
import java.util.List;
import java.util.Set;

public class EmailCounterJava {

    public static String normalizeEmailJava(String email) {
        String[] parts = email.split("@");
        if (parts.length != 2) {
            return ""; // Invalid email format
        }
        String username = parts[0];
        String domain = parts[1];

        StringBuilder normalizedUsername = new StringBuilder(); // Строим нормализованное имя
        for (char c : username.toCharArray()) { // Перебираем символы имени
            if (c == '.') {
                continue;
            } else if (c == '+' || c == '*') {
                break;
            } else {
                normalizedUsername.append(c);
            }
        }
        return normalizedUsername.toString() + "@" + domain;
    }

    public static int countUniqueEmailsJava(List<String> emails) { // Функция подсчета уникальных email
        Set<String> uniqueEmails = new HashSet<>(); // Множество для уникальных email
        for (String email : emails) { // Перебираем список email
            uniqueEmails.add(normalizeEmailJava(email)); // Нормализуем и добавляем в множество
        }
        return uniqueEmails.size();
    }

    public static void main(String[] args) {
        List<String> emails1 = List.of("mar.pha+science@corp.nstu.ru", "marpha+scie.nce@corp.nstu.ru", "marph.a+s.c.i.e.n.c.e+@corp.nstu.ru");
        System.out.println("Example 1 Output: " + countUniqueEmailsJava(emails1));

        List<String> emails2 = List.of("mar.pha+science@co.rp.nstu.ru", "marpha+scie.nce@corp.nstu.ru", "marph.a+s.c.i.e.n.c.e+@corp.nstu.ru");
        System.out.println("Example 2 Output: " + countUniqueEmailsJava(emails2));
    }
}
